squiggy/api/whiteboard_controller.py

```diff
@@ -0,0 +1,110 @@
+# POST: '/whiteboards/<whiteboard_id>/export/asset',
+# POST: '/whiteboards/<whiteboard_id>/restore',
+
+
+@app.route('/api/whiteboard/<whiteboard_id>')
+@login_required
+def get_whiteboard(whiteboard_id):
+    whiteboard = Whiteboard.find_by_id(whiteboard_id=whiteboard_id)
+    app.logger.warn(f'current_user: {current_user.course}')
+    if whiteboard and can_view_whiteboard(user=current_user, whiteboard=whiteboard):
+        return tolerant_jsonify(whiteboard.to_api_json())
+    else:
+        raise ResourceNotFoundError(f'No asset found with id: {whiteboard_id}')
+
+
+@app.route('/api/whiteboards', methods=['POST'])
+@login_required
+def get_whiteboards():
+    # TODO
+    # params = request.get_json()
+    # order_by = _get(params, 'orderBy', 'recent')
+    # offset = params.get('offset')
+    # limit = params.get('limit')
+    return tolerant_jsonify([w.to_api_json() for w in Whiteboard.get_whiteboards()])
+
+
+@app.route('/api/whiteboard/create', methods=['POST'])
+@login_required
+def create_whiteboard():
+    if not current_user.course:
+        raise BadRequestError('Course data not found')
+    params = request.get_json() or request.form
+    title = params.get('title')
+    if not title:
+        raise BadRequestError('Title is required.')
+
+    whiteboard = Whiteboard.create(
+        course_id=current_user.course.id,
+        title=title,
+        users=[User.find_by_id(current_user.get_id())],
+    )
+    return tolerant_jsonify(whiteboard.to_api_json())
+
+
+@app.route('/api/whiteboard/<whiteboard_id>/delete', methods=['DELETE'])
+@login_required
+def delete_whiteboard(whiteboard_id):
+    whiteboard = Whiteboard.find_by_id(whiteboard_id) if whiteboard_id else None
+    if not whiteboard:
+        raise ResourceNotFoundError('Whiteboard not found.')
+    if not can_update_asset(asset=whiteboard, user=current_user):
+        raise BadRequestError('To delete this asset you must own it or be a teacher or admin in the course.')
+
+    Whiteboard.delete(whiteboard_id=whiteboard_id)
+    return tolerant_jsonify({'message': f'Whiteboard {whiteboard_id} deleted'}), 200
+
+
+@app.route('/api/whiteboard/update', methods=['POST'])
+@login_required
+def update_whiteboard():
+    params = request.get_json()
+    whiteboard_id = params.get('whiteboardId')
+    title = params.get('title')
+    whiteboard = Whiteboard.find_by_id(whiteboard_id) if whiteboard_id else None
+    if not whiteboard or not title:
+        raise BadRequestError('Whiteboard update requires a valid ID and title.')
+    if not can_update_asset(asset=whiteboard, user=current_user):
+        raise BadRequestError('To update an asset you must own it or be a teacher in the course.')
+
+    whiteboard = Whiteboard.update(whiteboard_id=whiteboard_id, title=title)
+    return tolerant_jsonify(whiteboard.to_api_json(user_id=current_user.get_id()))
+
+
+def _get(_dict, key, default_value=None):
+    return _dict[key] if key in _dict else default_value
```

Comment:
I didn't mean to include `whiteboard_controller.py` in this PR but no problem. The TODOs are present. The associated tests are still cooking.

Reply:
Works for me.